platform/linux-generic/arch/aarch64/odp_random.h

```diff
@@ -0,0 +1,74 @@
+/* Copyright (c) ARM Limited
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier:     BSD-3-Clause
+ */
+
+#ifndef ODP_AARCH64_RANDOM_H_
+#define ODP_AARCH64_RANDOM_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <odp/api/spec/random.h>
+#include <odp/autoheader_internal.h>
+
+#include <stdint.h>
+
+#define RNDR_OUTPUT_LEN      8
+
+odp_random_kind_t _odp_random_max_kind_generic(void);
+int32_t _odp_random_true_data_generic(uint8_t *buf, uint32_t len);
+int32_t _odp_random_crypto_data_generic(uint8_t *buf, uint32_t len);
+
+static inline int _odp_random_max_kind(void)
+{
+	if (_ODP_OPENSSL_RAND) {
+#ifdef __ARM_FEATURE_RNG
+		return ODP_RANDOM_TRUE;
+#else
+		return ODP_RANDOM_CRYPTO;
+#endif
+	} else {
+		return ODP_RANDOM_BASIC;
+	}
+}
+
+static inline int _odp_rndr(unsigned long *v)
+{
+	int pass;
+
+	__asm__ volatile("mrs   %0, s3_3_c2_c4_0\n\t"  /* RNDR */
+			 "cset  %w[pass], ne"
+			 : "=r" (*v), [pass] "=r" (pass)
+			 :
+			 : "cc");
+
+	return pass;
+}
+
+static inline int32_t _odp_random_true_data(uint8_t *buf, uint32_t len)
+{
+#ifdef __ARM_FEATURE_RNG
+	if (_ODP_OPENSSL_RAND) {
+		while (!_odp_rndr((unsigned long *)buf))
+			;
+
+		return RNDR_OUTPUT_LEN;
+	}
+#endif
+
+	return _odp_random_true_data_generic(buf, len);
+}
+
+static inline int32_t _odp_random_crypto_data(uint8_t *buf, uint32_t len)
+{
+	return _odp_random_crypto_data_generic(buf, len);
```

Comment:
From `odp_random_max_kind()` spec: `Implementations support the returned max kind and all kinds weaker than it.` So if `ODP_RANDOM_TRUE` is supported, `ODP_RANDOM_CRYPTO` should also be supported. The same `_odp_rndr()` function could be used for the implementation.

Reply:
Please check the comment above

Reply:
Or, similarly to what we decided on x86 with RDRAND and RDSEED, since RNDRRS is seeded every time from TRNG and RNDR only occasionally, RNDR should probably be used only for ODP_RANDOM_CRYPTO, and RNDRRS for ODP_RANDOM_TRUE.

Reply:
Hi @JereLeppanen, this thought had occurred to me to as well and I did a bit of research on it to see if RNDR ~= RDRAND and RNDRRS ~= RDSEED for the purpose of this PR. I found this discussion on the LKML which exactly discusses if this association is valid: https://lore.kernel.org/linux-arm-kernel/6e75d7b9-1c30-adab-bb74-1aaaa4e98ad4@linaro.org/
If you still think that we should proceed with RNDR for `ODP_RANDOM_CRYPTO`, and RNDRRS for `ODP_RANDOM_TRUE`, I'll make the changes in v2

Reply:
Yes, I saw that, and it looks like that discussion was never fully resolved, at least not explicitly. As far as I can tell, the "RNDR ~= RDRAND and RNDRRS ~= RDSEED" comment in that thread is correct.

Reply:
Okay, let me make these changes in v2. Thanks Jere!